StringFormatterBench/Program.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System.Text.Formatting;
+using BenchmarkDotNet.Attributes;
+using BenchmarkDotNet.Columns;
+using BenchmarkDotNet.Configs;
+using BenchmarkDotNet.Diagnostics.Windows;
+using BenchmarkDotNet.Jobs;
+using BenchmarkDotNet.Running;
+
+namespace StringFormatterBench
+{
+    public class Config : ManualConfig
+    {
+        public Config()
+        {
+            Add(Job.LegacyJitX64, Job.RyuJitX64);
+            //Add(Job.Mono);
+            Add(StatisticColumn.Min);
+            Add(StatisticColumn.Max);
+            Add(new MemoryDiagnoser());
+        }
+    }
+
+    [Config(typeof(Config))]
+    public class StringFormat
+    {
+        StringBuffer _sb;
+        static readonly string formatTest = "Foo {0,13:e12} and bar!! {1,-15:P}bah";
+        const double v1 = 13.934939;
+        const double v2 = 0;
+
+        [Setup]
+        public void Setup()
+        {
+            _sb = new StringBuffer();
+        }
+
+        [Benchmark(Baseline = true)]
+        public string Baseline()
+        {
+            return string.Format(formatTest, v1, v2);
```

Comment:
For maximum fairness you should probably do Setup() work for this case as well, since the first call to string.Format() is going to do a StringBuilder allocation.